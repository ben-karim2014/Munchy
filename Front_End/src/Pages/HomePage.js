//This is a public page route

import React, { Component, Fragment } from "react";
import HeaderApp from "../Components/Header";
//import Carousel from "../Components/Slide";
import "react-responsive-carousel/lib/styles/carousel.min.css";
import AliceCarousel from 'react-alice-carousel';
import image1 from "../images/img1.jpg"
import image2 from "../images/img3.jpg"
import image3 from "../images/img4.jpg"
import image4 from "../images/img5.jpg"
import { useState } from 'react';
import {
  Carousel,
  CarouselItem,
  CarouselControl,
  CarouselIndicators,
  CarouselCaption
} from 'reactstrap'; 

const items = [
  {
    src: image1,
    altText: '',
    caption: ''
  },
  {
    src: image2,
    altText: '',
    caption: ''

  },
  {
    src: image3,
    altText: 'Slide 3',
    caption: ''

  },
  {
    src: image4,
    altText: 'Slide 4',
    caption: ''

  }
];
const HomePage = (props) =>  {
  
  const [activeIndex, setActiveIndex] = useState(0);
  const [animating, setAnimating] = useState(false);

  const next = () => {
    if (animating) return;
    const nextIndex = activeIndex === items.length - 1 ? 0 : activeIndex + 1;
    setActiveIndex(nextIndex);
  }

  const previous = () => {
    if (animating) return;
    const nextIndex = activeIndex === 0 ? items.length - 1 : activeIndex - 1;
    setActiveIndex(nextIndex);
  }

  const goToIndex = (newIndex) => {
    if (animating) return;
    setActiveIndex(newIndex);
  }

  const slides = items.map((item) => {
    return (
      <CarouselItem
        onExiting={() => setAnimating(true)}
        onExited={() => setAnimating(false)}
        key={item.src}
      >
        <img src={item.src} alt={item.altText} />
        <CarouselCaption captionText={item.caption} captionHeader={item.caption} />
      </CarouselItem>
    );
  });

    return (
      <Fragment>
        <HeaderApp />
        <Carousel
        activeIndex={activeIndex}
        next={next}
        previous={previous}
      >
        <CarouselIndicators items={items} activeIndex={activeIndex} onClickHandler={goToIndex} />
        {slides}
        <CarouselControl direction="prev" directionText="Previous" onClickHandler={previous} />
        <CarouselControl direction="next" directionText="Next" onClickHandler={next} />
      </Carousel>

      </Fragment>
    );
  
}

export default HomePage;
