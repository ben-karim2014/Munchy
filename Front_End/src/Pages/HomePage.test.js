import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from "react-redux";
import store from "../store";

import HomePage from './HomePage';

it('renders correctly', () => {
    const tree = renderer.create(<Provider store={store}><HomePage /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});