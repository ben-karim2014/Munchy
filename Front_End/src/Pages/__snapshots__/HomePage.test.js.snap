// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="bottomBorder"
    data-testid="upper"
  >
    <nav
      className="header navbar navbar-expand-sm navbar-dark"
      data-testid="upper2"
    >
      <a
        href="/"
      >
        <img
          alt=""
          className="logo"
          src="munchy_logo.png"
        />
      </a>
      <button
        aria-label="Toggle navigation"
        className="navbar-toggler"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        data-testid="upper3"
        style={Object {}}
      >
        <div
          className="header"
        >
          <ul
            className="navbar-header center navbar-nav"
          >
            <li
              className="nav-list nav-item"
            >
              <a
                className="nav-link"
                href="/"
                onClick={[Function]}
              >
                Home
              </a>
            </li>
            <li
              className="nav-list nav-item"
            >
              <a
                className="nav-link"
                href="/login"
                onClick={[Function]}
              >
                Sign in
              </a>
            </li>
            <li
              className="nav-list nav-item"
            >
              <a
                className="nav-link"
                href="/signup"
                onClick={[Function]}
              >
                Sign up
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </div>,
  <div
    className="carousel slide"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
  >
    <ol
      className="carousel-indicators"
    >
      <li
        className="active"
        onClick={[Function]}
      />
      <li
        className=""
        onClick={[Function]}
      />
      <li
        className=""
        onClick={[Function]}
      />
      <li
        className=""
        onClick={[Function]}
      />
    </ol>
    <div
      className="carousel-inner"
    >
      <div
        className="carousel-item active"
      >
        <img
          alt=""
          src="img1.jpg"
        />
        <div
          className="carousel-caption d-none d-md-block"
        >
          <h3>
            
          </h3>
          <p>
            
          </p>
        </div>
      </div>
      <div
        className="carousel-item"
      >
        <img
          alt=""
          src="img3.jpg"
        />
        <div
          className="carousel-caption d-none d-md-block"
        >
          <h3>
            
          </h3>
          <p>
            
          </p>
        </div>
      </div>
      <div
        className="carousel-item"
      >
        <img
          alt="Slide 3"
          src="img4.jpg"
        />
        <div
          className="carousel-caption d-none d-md-block"
        >
          <h3>
            
          </h3>
          <p>
            
          </p>
        </div>
      </div>
      <div
        className="carousel-item"
      >
        <img
          alt="Slide 4"
          src="img5.jpg"
        />
        <div
          className="carousel-caption d-none d-md-block"
        >
          <h3>
            
          </h3>
          <p>
            
          </p>
        </div>
      </div>
    </div>
    <a
      className="carousel-control-prev"
      onClick={[Function]}
      role="button"
      style={
        Object {
          "cursor": "pointer",
        }
      }
      tabIndex="0"
    >
      <span
        aria-hidden="true"
        className="carousel-control-prev-icon"
      />
      <span
        className="sr-only"
      >
        Previous
      </span>
    </a>
    <a
      className="carousel-control-next"
      onClick={[Function]}
      role="button"
      style={
        Object {
          "cursor": "pointer",
        }
      }
      tabIndex="0"
    >
      <span
        aria-hidden="true"
        className="carousel-control-next-icon"
      />
      <span
        className="sr-only"
      >
        Next
      </span>
    </a>
  </div>,
]
`;
